operator/controllers/btpoperator_controller.go

```diff
@@ -199,39 +404,63 @@ func (r *BtpOperatorReconciler) HandleInitialState(ctx context.Context, cr *v1al
 	return r.UpdateBtpOperatorStatus(ctx, cr, types.StateProcessing, Initialized, "Initialized")
 }
 
+func (r *BtpOperatorReconciler) CheckIfReasonIsSet(cr *v1alpha1.BtpOperator, reason Reason) bool {
+	var condition *v1.Condition
+	if len(cr.Status.Conditions) > 0 {
+		condition = cr.Status.Conditions[0]
+	}
+	return condition != nil && condition.Reason == string(reason)
+}
+
 func (r *BtpOperatorReconciler) HandleProcessingState(ctx context.Context, cr *v1alpha1.BtpOperator) error {
 	logger := log.FromContext(ctx)
-	logger.Info("Handling Processing state")
```

Comment:
maybe it's worth to keep this log and provide separate logs for provisioning and update operations